transaction/_transaction.py

```diff
@@ -409,6 +416,67 @@ def _callAfterCommitHooks(self, status=True):
         self._after_commit = []
         self._before_commit = []
 
+    def getBeforeAbortHooks(self):
```

Comment:
I feel like there's enough duplication across these four different hooks that some meta-programming to reduce duplication might be called for. That can be a separate PR though.